satpy/scene.py

```diff
@@ -1023,6 +1031,52 @@ def show(self, dataset_id, overlay=None):
             img.show()
         return img
 
+    def to_geoviews(self, gvtype=gv.Image, datasets=None, kdims=None, vdims=None, dynamic=False):
+        """Convert satpy Scene to geoviews.
+
+        Parameters
+        ----------
+        gvtype : gv plot type, optional, default gv.Image
+            One of gv.Image, gv.LineContours, gv.FilledContours, gv.Points
+            See Geoviews documentation for details.
+        datasets : list of string
+        kdims : list of str, optional
+            Key dimensions. See geoviews documentation for more information.
+        vdims : list of str, optional
+            Value dimensions. See geoviews documentation for more information.
+            If not given defaults to first data variable
+        dynamic : boolean, optional, default False
+
+        Returns
+        -------
+        geoviews object
+
+        Todo
+        ----
+        - better handling of projection information in datasets which are
+          to be passed to geoviews
+        """
+        ds = scn_to_xrds(self, datasets)
+
+        if vdims is None:
+            # by default select first data variable as display variable
+            vdims = ds.data_vars[ds.data_vars.keys()[0]].name
+
+        if hasattr(ds, "area"):
```

Comment:
Pyresample `AreaDefinition` objects have a `to_cartopy_crs()` that could be used to replace this.

Reply:
This wouldn't help with `SwathDefinition` objects with plain longitude/latitude arrays, but I'm not familiar with how geoviews would handle that anyway.

Reply:
I would have thought that the object returned by `to_cartopy_crs` is compatible. At least it is compatible with cartopy. There is a cartopy PR that I'm working on to make this more cartopy-friendly, but it has been lacking feedback from the cartopy maintainers due to time constraints from the sound of it.

Regarding swath definitions: ok. I don't know anything about QuadMeshs as far as geoviews thinks of them.

Reply:
Had another look at it and uses to_cartopy_crs() now.